openquake/calculators/postproc/compute_rtgm.py

```diff
@@ -326,6 +340,284 @@ def get_asce41(dstore, mce, facts):
             'BSE1N_S1': BSE1N_S1}
 
 
+def _get_label(imt):
+    imtlab = imt if imt == 'PGA' else imt.replace(')', 's)')
+    comp = 'Geom. mean' if imt == 'PGA' else 'Max. comp.'
+    return imtlab + ' - ' + comp
+
+
+def plot_meanHCs_afe_RTGM(imls, AFE, UHGM_RP, afe_RP, RTGM, afe_RTGM,
+                          imt_list):
+    plt.figure(figsize=(12, 9))
+    plt.rcParams.update({'font.size': 16})
+    colors = mpl.colormaps['viridis'].reversed()._resample(3)
+    patterns = ['-', '--', ':']
+    for i, imt in enumerate(imt_list):
+        lab = _get_label(imt)
+        plt.loglog(imls[i], AFE[i], color=colors(i), linestyle=patterns[i],
+                   label=lab, linewidth=3, zorder=1)
+        if i == 2:
+            plt.loglog([RTGM[i]], [afe_RTGM[i]], 'ko',
+                       label='Probabilistic MCE',  linewidth=2,
+                       markersize=10, zorder=3)
+        else:
+            plt.loglog([RTGM[i]], [afe_RTGM[i]], 'ko',
+                       linewidth=2, markersize=10, zorder=3)
+        plt.loglog([np.min(imls[i]), RTGM[i]], [afe_RTGM[i], afe_RTGM[i]],
+                   'darkgray', linestyle='--', linewidth=1)
+        plt.loglog([RTGM[i], RTGM[i]], [0, afe_RTGM[i]], 'darkgray',
+                   linestyle='--', linewidth=1)
+
+    plt.grid('both')
+    plt.legend(fontsize=13)
+    plt.xlabel('Acceleration (g)', fontsize=20)
+    plt.ylabel('Annual frequency of exceedance', fontsize=20)
+    plt.legend(loc="best", fontsize='16')
+    plt.ylim([10E-6, 1.1])
+    plt.xlim([0.01, 4])
+    bio = io.BytesIO()
+    plt.savefig(bio, format='png', bbox_inches='tight')
+    plt.clf()
+    return Image.open(bio)
+
+
+def _find_afe_target(imls, afe, sa_target):
+    # find the target afe (or poe) for a given acceleration
+    if len(imls) != len(afe):
+        afe.extend([1E-15] * (len(imls) - len(afe)))
+    f = interpolate.interp1d(np.log(imls), np.log(afe))
+    afe_target = np.exp(f(np.log(sa_target)))
+    return afe_target
+
+
+def disaggr_by_src(dstore, imtls):
+    imtls_dict = {}
+    for imt, imls in imtls.items():
+        imls = [iml for iml in imls]
+        imtls_dict[imt] = imls
+    # get info : specific to disagg by src
+    df = dstore['mean_rates_by_src'].to_dframe().set_index('src_id')
+    grouped_m = df.groupby(['src_id', 'site_id', 'imt']).agg(
+        {"value": lambda x: list(x)}).reset_index()
```

Comment:
`lambda x: list(x)` can be replaced with `list`

Reply:
replaced